### DISCUSSION TITLE
Export cue file

### DISCUSSION THREAD

**Discussion by jimbo1jimbo** [yermak/AudioBookConverter#471] [Ideas]
Would it be possible to export a cue file from an existing m4b file?

Maybe add "cue" as a "Format" option ...

**Comment by yermak** [comment]
@jimbo1jimbo 
What is usecase scenario for this? You will not get better quality if source if m4b... Are you going to record CDs? Probably most of CD players do support mp3

**Comment by jimbo1jimbo** [comment]
Thanks for the quick reply.

I have quite a few old M4B files that have "chapter" marks, but they're not
quite correct. Example:

1 = Chapter 1
2 = Chapter 4
3 = Chapter 6
4 = Chapter 7
5 = Chapter 10

I can find the missing chapter points reasonably easily. And I *could*
therefore create a CUE file from scratch, with the details of all 12
chapters.

But it would be far easier if I had the known chapters in a CUE file
(exported from AudioBookConverter), and just had to add the details of the
7 missing chapters.

**Comment by yermak** [comment]
Ah, got it.
So, you don't need cue as such. You need to custom editing of chapter marks.
Unfortunately I can't think out of the box how to do it with existing version.

Right solution would be: go to chapters mode, and then right-click on track within chapter, select sub-track... unfortunately currently program supports creating sub-tracks with specific intervals, but not at specific point. What you would like to see is break file into sub-track AT SPECIFIC TIME, so it would create you 2 chapters from single file. then you could repeat it several times to create all your missing chapters.
Alternative way - to allow custom edit of all chapters information, i.e. upload cue file which will overwrite all chapters information. Honestly, i don't think many people will use it.

I keep in mind this idea, migth add it in future...

**Comment by jimbo1jimbo** [comment]
Thanks again for the quick reply

Agree that editing chapters is the ideal solution, but seems very difficult
to engineer

Understand that exporting CUE is not massive priority, but much easier than
above (all data known, and required format known).

Thanks anyway.

---

As an aside, I thought I'd try to use the Split by chapters functionality,
so that I could then edit in GoldWave to create correct individual chapter
files, for re-merging in AudioBookConverter.

However, I have a real problem with doing that - the file names assigned to
the split files do not match the names in the Chapters table:

Book 1 Chapter 1 is indeed Chapter 1
Book 1 Chapter 2 is actually Chapter 4
Book 1 Chapter 3 is actually Chapter 2
and so on.
